sdk/eventhub/event-hubs/src/connectionContext.ts

```diff
@@ -48,6 +51,24 @@ export interface ConnectionContext extends ConnectionContextBase {
    * the underlying amqp connection for the EventHub Client.
    */
   managementSession?: ManagementClient;
+
+  /**
+   * Indicates whether the connection is in the process of closing.
+   */
+  isConnectionClosing(): boolean;
+  /**
+   * Resolves once the connectionContext is ready to be opened.
+   */
+  readyToOpen(): Promise<void>;
+  /**
+   * Resolves once the context's connection emits a `disconnected` event.
+   */
+  waitForConnectionDisconnected(): Promise<void>;
+  /**
+   * Resolves once the connection has finished being reset.
+   * Connections are reset as part of reacting to a `disconnected` event.
+   */
+  waitForConnectionReset(): Promise<void>;
```

Comment:
Both `waitForConnectionReset` and  `waitForConnectionDisconnected ` are used only inside `readyToOpen()`. So, why should they be public methods on the connection context?